temporian/core/data/dtype.py

```diff
@@ -40,3 +41,25 @@ def MissingValue(dtype: DType) -> Any:
         return ""
     else:
         raise ValueError(f"Non implemented type {dtype}")
+
+
+def python_type_to_temporian_dtype(python_type: Any) -> DType:
+    """
+    Convert a Python type to its corresponding Temporian dtype.
+    Args:
+        python_type: The Python type to be converted.
+    Returns:
+        The corresponding Temporian dtype.
+    Raises:
+        ValueError: If the input Python type is not supported.
+    """
+    if issubclass(python_type, float):
+        return FLOAT64  # Assuming default float type is FLOAT64
+    if issubclass(python_type, int):
+        return INT64  # Assuming default int type is INT64
+    if issubclass(python_type, str):
+        return STRING
+    if issubclass(python_type, bool):
+        return BOOLEAN
```

Comment:
why compare using `issubclass` instead of  using `python_type is float` (etc.)? do we also want to support custom python types?